util/eventbus/eventbustest/eventbustest.go

```diff
@@ -201,3 +201,41 @@ func eventFilter(f any) filter {
 		return fixup(fv.Call([]reflect.Value{args[0].Elem()}))
 	}).Interface().(filter)
 }
+
+// Injector holds a map with [eventbus.Publisher], tied to an [eventbus.Client]
+// for testing purposes.
+type Injector struct {
+	client     *eventbus.Client
+	publishers map[reflect.Type]any
+	// The value for a key is an *eventbus.Publisher[T] for the corresponding type.
+}
+
+// NewInjector constructs an [Injector] that can be used to inject events into
+// the the stream of events used by code under test. After construction the
+// caller may use [Inject] to insert events into the bus.
+func NewInjector(t *testing.T, b *eventbus.Bus) *Injector {
+	inj := &Injector{
+		client:     b.Client(t.Name()),
+		publishers: make(map[reflect.Type]any),
+	}
+	t.Cleanup(inj.client.Close)
+
+	return inj
+}
+
+// Inject inserts events of T onto an [eventbus.Bus]. If an [eventbus.Publisher]
+// for the type does not exist, it will be initialized lazily. Calling inject is
+// synchronous, and the event will as such have been published to the eventbus
+// by the time the function returns.
+func Inject[T any](inj *Injector, event T) {
+	eventType := reflect.TypeFor[T]()
+
+	pub, ok := inj.publishers[eventType]
+	if !ok {
+		pub = eventbus.Publish[T](inj.client)
+		inj.publishers[eventType] = pub
+	}
+
+	castPub := pub.(*eventbus.Publisher[T])
+	castPub.Publish(event)
```

Comment:
Fine as-written, but could also be:
```suggestion
	pub.(*eventbus.Publisher[T]).Publish(event)
```